### PR TITLE
feat(frontend): add exporerUrl to BTC networks [GIX-2795]

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The goal is to extend BTC networks with already defined explorer URLs. It will be used in the BTC transaction detail modal.

### CODE REVIEW

**Review by AntonioVentilii-DFINITY — COMMENTED** [review]
Just to simplify the code, we have already `EthereumAppMetadata`: can we "merge" the two types and have something like `NetworkAppMetadata`? (that should include the original optionality/non-optionality of both)

**Comment by DenysKarmazynDFINITY** [comment]
@AntonioVentilii-DFINITY good idea, it's done. `explorerUrl` is required in `NetworkMetadataApp` but I made it optional just for BTC to be compatible with regtest network type (no explorer there).

File: src/frontend/src/btc/types/network.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import type { Network, NetworkAppMetadata } from '$lib/types/network';
+
+export type BitcoinNetwork = Network & Partial<Pick<NetworkAppMetadata, 'explorerUrl'>>;
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/btc/types/network.ts:3]: For now, I think it is safe to remove the Pick part, right? Just an observation, unless you plan already on adding more things there that will be non-optional

**Review by AntonioVentilii-DFINITY — APPROVED** [review]
LGTM! Small observation

**Pull request merged**